### PR TITLE
master <- develop: find journey form

### PR DESCRIPTION
**Pull request by blablanumerodeux**

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -1,33 +1,33 @@
 <div class="example-container" [class.example-is-mobile]="mobileQuery.matches">
 
   <mat-toolbar color="primary" class="example-toolbar">
-    <span>Dakar</span>
+    <span>Dakar </span>
     <button mat-icon-button color="warn" (click)="snav.toggle()">
```
Comment (shulolo) [inline src/app/app.component.html]: snav is not a explicit name

File: src/app/app.component.html [hunk 2]
```diff
@@ -1,33 +1,33 @@
...
       <mat-icon aria-label="Example icon-button with a menu icon">menu</mat-icon>
     </button>
-    <h1 class="example-app-name">Responsive App</h1>
+    <h1 class="example-app-name">{{title}}</h1>
   </mat-toolbar>
 
   <mat-sidenav-container class="example-sidenav-container"
                          [style.marginTop.px]="mobileQuery.matches ? 56 : 0">
```
Comment (shulolo) [inline src/app/app.component.html]: same for mobileQuery name

File: src/app/app.component.html [hunk 3]
```diff
@@ -1,33 +1,33 @@
...
 
-    <mat-sidenav #snav [mode]="mobileQuery.matches ? 'over' : 'side'"
+    <mat-sidenav opened #snav [mode]="mobileQuery.matches ? 'over' : 'side'"
```
Comment (shulolo) [inline src/app/app.component.html]: what is "opened" ?

File: src/app/app.component.html [hunk 4]
```diff
@@ -1,33 +1,33 @@
...
                  [fixedInViewport]="mobileQuery.matches" fixedTopGap="56">
```
Comment (shulolo) [inline src/app/app.component.html]: what is fixedToGap?

File: src/app/app.component.ts [hunk 5]
```diff
@@ -12,9 +12,7 @@ export class AppComponent implements OnDestroy {
 
   mobileQuery: MediaQueryList;
 
-  menu = ['find-journey', 'create journey', 'list journey'];
-
-  fillerNav = this.menu;
+  fillerNav = ['find-journey', 'create journey', 'list journey'];
```
Comment (shulolo) [inline src/app/app.component.ts]: I don't think that is a good idea to use the same variable for the link and for the name displayed for the link.

File: src/app/app.component.ts [hunk 6]
```diff
@@ -12,9 +12,7 @@ export class AppComponent implements OnDestroy {
...
 
   fillerContent = Array.from({length: 50}, () =>
```
Comment (shulolo) [inline src/app/app.component.ts]: You can change the content to put for example "DaKar, travel as you imagine", or "DaKar, compose your journey of your dreams", or "DaKar, just take your valise"....

File: src/app/components/find-journey/find-journey.component.html [hunk 7]
```diff
@@ -1,6 +1,20 @@
 <div style="text-align:center">
   <h1>
+    <!--lets try to fetch the title from a son-->
```
Comment (shulolo) [inline src/app/components/find-journey/find-journey.component.html]: I don't understand the comment.

File: src/app/components/find-journey/find-journey.component.html [hunk 8]
```diff
@@ -1,6 +1,20 @@
...
     Welcome to {{ title }}!
   </h1>
 </div>
+<form class="example-form" (ngSubmit)="onSubmit()">
```
Comment (shulolo) [inline src/app/components/find-journey/find-journey.component.html]: for your information there is another way to create a form, we can discuss it later.

File: src/app/components/find-journey/find-journey.component.html [hunk 9]
```diff
@@ -1,6 +1,20 @@
...
+  <mat-form-field class="example-full-width">
+    <input matInput placeholder="Favorite food" value="Sushi">
```
Comment (shulolo) [inline src/app/components/find-journey/find-journey.component.html]: matinput?

File: src/app/components/find-journey/find-journey.component.ts [hunk 10]
```diff
@@ -8,15 +14,79 @@ import {Component, OnInit} from '@angular/core';
...
+    this.journey = new Journey();
+    this.journeysFound = [];
   }
 
   onSubmit() {
-    console.log('test');
+    console.log('id searched: ', this.id.value);
+    console.log('destination searched: ', this.destination.value);
+
+    this.searchById();
```
Comment (shulolo) [inline src/app/components/find-journey/find-journey.component.ts]: I forgot but now we can do a search by destination and/or price so I purpose you to do the research with that data.

File: src/app/components/find-journey/find-journey.component.ts [hunk 11]
```diff
@@ -8,15 +14,79 @@ import {Component, OnInit} from '@angular/core';
...
   }
 
+  searchById() {
```
Comment (shulolo) [inline src/app/components/find-journey/find-journey.component.ts]: It should on a service

File: src/app/components/find-journey/find-journey.component.ts [hunk 12]
```diff
@@ -8,15 +14,79 @@ import {Component, OnInit} from '@angular/core';
...
+        query find{
+          findJourneyById(id: "tt") {
+            id,
+            destination
+          }
+        }
+      `
+    }).pipe(shareReplay(1));
+
+    this.destination$ = source$.pipe(map(result => result.data && result.data['findJourneyById']));
```
Comment (shulolo) [inline src/app/components/find-journey/find-journey.component.ts]: I don't understand this line

File: src/app/components/find-journey/find-journey.component.ts [hunk 13]
```diff
@@ -8,15 +14,79 @@ import {Component, OnInit} from '@angular/core';
...
+    this.destination$.subscribe(
+      (response) => {
+        console.log(response);
```
Comment (shulolo) [inline src/app/components/find-journey/find-journey.component.ts]: remove the console.log expect if there are useful

File: src/app/components/journey/journey.component.ts [hunk 14]
```diff
@@ -1,54 +1,17 @@
...
 import {Journey} from '../../models/Journey';
-import gql from 'graphql-tag';
-import {map, shareReplay} from 'rxjs/operators';
-import {Apollo} from 'apollo-angular';
-import {Observable} from 'rxjs';
 
 @Component({
   selector: 'app-journey',
   templateUrl: './journey.component.html',
   styleUrls: ['./journey.component.scss']
```
Comment (shulolo) [inline src/app/components/journey/journey.component.ts]: What is this component?

File: src/app/app.component.html (hunk 3 above)
Comment (blablanumerodeux) [reply src/app/app.component.html]: it's an attribute of mat-sidenav proposed by Angular Material

File: src/app/app.component.html (hunk 4 above)
Comment (blablanumerodeux) [reply src/app/app.component.html]: no idea

File: src/app/app.component.ts (hunk 5 above)
Comment (blablanumerodeux) [reply src/app/app.component.ts]: right we need a refacto on that

File: src/app/components/find-journey/find-journey.component.html (hunk 8 above)
Comment (blablanumerodeux) [reply src/app/components/find-journey/find-journey.component.html]: discussed

File: src/app/components/find-journey/find-journey.component.html (hunk 9 above)
Comment (blablanumerodeux) [reply src/app/components/find-journey/find-journey.component.html]: it was in the example. 
I didn't check the doc yet.

File: src/app/components/find-journey/find-journey.component.ts [hunk 15]
```diff
@@ -8,15 +14,79 @@ import {Component, OnInit} from '@angular/core';
...
+export class FindJourneyComponent {
 
-  constructor() { }
+  journey: Journey;
+  journeysFound: Journey[];
+  destination = new FormControl('');
+  id = new FormControl('');
 
-  ngOnInit() {
+  destination$: Observable<any[]>;
```
[the remark this answers is not in the document]
Comment (blablanumerodeux) [reply src/app/components/find-journey/find-journey.component.ts]: it's a convention to indicate that it's an observable

File: src/app/components/find-journey/find-journey.component.ts (hunk 10 above)
Comment (blablanumerodeux) [reply src/app/components/find-journey/find-journey.component.ts]: that's my next step

File: src/app/components/find-journey/find-journey.component.ts (hunk 11 above)
Comment (blablanumerodeux) [reply src/app/components/find-journey/find-journey.component.ts]: done

File: src/app/components/find-journey/find-journey.component.ts (hunk 12 above)
Comment (blablanumerodeux) [reply src/app/components/find-journey/find-journey.component.ts]: we do a sort of 'pluck' here.
cause we don't need all the info returned by graphql

File: src/app/components/find-journey/find-journey.component.ts (hunk 13 above)
Comment (blablanumerodeux) [reply src/app/components/find-journey/find-journey.component.ts]: done

File: src/app/components/journey/journey.component.ts (hunk 14 above)
Comment (blablanumerodeux) [reply src/app/components/journey/journey.component.ts]: this is a component to display a single journey

File: src/app/components/find-journey/find-journey.component.html [hunk 16]
```diff
@@ -1,20 +1,19 @@
...
+  <mat-form-field class="input-box">
+    <textarea matInput [formControl]="id" placeholder="id"></textarea>
   </mat-form-field>
 
-  <br>
-  <mat-form-field class="example-full-width">
-    <textarea matInput placeholder="Leave a comment"></textarea>
+  <mat-form-field class="input-box">
+    <textarea matInput [formControl]="destination" placeholder="destination"></textarea>
   </mat-form-field>
```
[the remark this answers is not in the document]
Comment (blablanumerodeux) [reply src/app/components/find-journey/find-journey.component.html]: will be added later in another story

File: src/app/app.component.html [hunk 17]
```diff
@@ -1,33 +1,33 @@
 <div class="example-container" [class.example-is-mobile]="mobileQuery.matches">
 
   <mat-toolbar color="primary" class="example-toolbar">
-    <span>Dakar</span>
+    <span>Dakar </span>
```
[the remark this answers is not in the document]
Comment (blablanumerodeux) [reply src/app/app.component.html]: tickets created in the backlog

**Pull request merged**